src/sentry/search/events/builder/spans_indexed.py

```diff
@@ -46,8 +57,49 @@ def time_column(self) -> SelectType:
 class TopEventsSpanIndexedQueryBuilder(SpansIndexedQueryBuilderMixin, TopEventsQueryBuilder):
     config_class = SpansIndexedDatasetConfig
 
+    def default_filter_converter(self, search_filter: SearchFilter) -> WhereType | None:
+        validate_id_like_search_filter(search_filter)
+        return super().default_filter_converter(search_filter)
+
     @property
     def time_column(self) -> SelectType:
         return custom_time_processor(
             self.interval, Function("toUInt32", [Column("start_timestamp")])
         )
+
+
+SPAN_ID_FIELDS = {
+    "id": "Filter Span ID",
+    "span_id": "Filter Span ID",
+    "parent_span": "Filter Parent Span ID",
+    "parent_span_id": "Filter Parent Span ID",
+    "segment.id": "Filter Segment ID",
+    "segment_id": "Filter Segment ID",
+}
+
+UUID_FIELDS = {
+    "trace": "Filter Trace ID",
+    "trace_id": "Filter Trace ID",
+    "transaction.id": "Filter Transaction ID",
+    "transaction_id": "Filter Transaction ID",
+    "profile.id": "Filter Profile ID",
+    "profile_id": "Filter Profile ID",
+    "replay.id": "Filter Replay ID",
+    "replay_id": "Filter Replay ID",
+}
+
+
+def validate_id_like_search_filter(search_filter: SearchFilter):
+    name = search_filter.key.name
+
+    if label := SPAN_ID_FIELDS.get(name):
+        if search_filter.value.is_wildcard():
+            raise InvalidSearchQuery(WILDCARD_NOT_ALLOWED.format(label))
+        if not search_filter.value.is_span_id():
+            raise InvalidSearchQuery(INVALID_SPAN_ID.format(label))
+
+    if label := UUID_FIELDS.get(name):
+        if search_filter.value.is_wildcard():
+            raise InvalidSearchQuery(WILDCARD_NOT_ALLOWED.format(label))
+        if not search_filter.value.is_event_id():
+            raise InvalidSearchQuery(INVALID_ID_DETAILS.format(label))
```

Comment:
A block like this already exists in DiscoverQueryBuilder:127 should we move that into the base, and then override it in the SpanBuilder?